src/query/api/v1/handler/prom/read.go

```diff
@@ -109,6 +109,25 @@ func (h *readHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if !h.opts.instant {
```

Comment:
Hm, we may want to do this even for instant queries I believe in case the instant query timestamp is set to a long time in past (rather than `now()`) where resolution is more course than the range vector?

Reply:
Ah yeah, good catch. I got a little mixed up here in how I was thinking about the problem. Will remove these checks.